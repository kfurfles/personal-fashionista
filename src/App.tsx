
import React from 'react';
import { hot } from 'react-hot-loader/root';

import { Header } from './components/header/header'

interface Props {
   name: string
}

export function App (props: Props){
  const { name } = props;
  return (
    <div>
      <Header />
      <h1>Hello {name}</h1>
    </div>
  );
}

export default hot(App);
