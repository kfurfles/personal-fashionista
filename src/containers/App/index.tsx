
import React from 'react';
import { hot } from 'react-hot-loader/root';
import { 
  BrowserRouter as Router,
} from 'react-router-dom'

import { Header } from '../../components/header'
import { Routes } from '../../routes'

export function App (){
    return (
        <Router>
            <Header />
            <Routes />
        </Router>
    );
}

export default hot(App);
