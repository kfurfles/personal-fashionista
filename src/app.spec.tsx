import React from "react";
import { create } from "react-test-renderer";
import { ButtonComponent } from './App';

describe("Button component", () => {
  test("Matches the snapshot", () => {
    const button = create(<ButtonComponent text="teste" />);
    
    expect(button.toTree()?.props.text).toBe('teste')
    console.log(button.getInstance())
    expect(button.toJSON()).toMatchSnapshot();
  });
});

// Note: running cleanup afterEach is done automatically for you in @testing-library/react@9.0.0 or higher
// unmount and cleanup DOM after the test is finished.
