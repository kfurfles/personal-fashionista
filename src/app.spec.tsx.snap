// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Matches the snapshot 1`] = `
<div>
  <header>
    headers
    <i
      className="i-search"
    />
  </header>
  <h1>
    Hello 
    Jullian
  </h1>
</div>
`;
